### PR TITLE
Rewrite automation test script in python

### PR DESCRIPTION
**Pull request by mksanger**

### CODE REVIEW

**Comment by kjsanger** [comment]
What's the best way to run these locally? I've tried `cd`ing to `gitlab_test` and running `pytest`:

```
gitlab_test$ pytest
================================================================= test session starts =================================================================
platform linux -- Python 3.8.3, pytest-5.4.1, py-1.9.0, pluggy-0.13.1
rootdir: /home/keith/dev/work/cloud/npg_conda/tools/automation/gitlab_test, inifile: pytest.ini, testpaths: tests
plugins: it-0.1.4
collected 0 items / 2 errors                                                                                                                          

======================================================================= ERRORS ========================================================================
_______________________________________________________ ERROR collecting tests/channel_test.py ________________________________________________________
ImportError while importing test module '/home/keith/dev/work/cloud/npg_conda/tools/automation/gitlab_test/tests/channel_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/channel_test.py:6: in <module>
    from channel import Channel
E   ModuleNotFoundError: No module named 'channel'
_______________________________________________________ ERROR collecting tests/package_test.py ________________________________________________________
ImportError while importing test module '/home/keith/dev/work/cloud/npg_conda/tools/automation/gitlab_test/tests/package_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/package_test.py:5: in <module>
    from conda.cli.python_api import Commands, run_command
E   ModuleNotFoundError: No module named 'conda'
=============================================================== short test summary info ===============================================================
ERROR tests/channel_test.py
ERROR tests/package_test.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================================== 2 errors in 0.08s ==================================================================
```

File: tools/automation/gitlab_test/pytest.ini [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[pytest]
+testpaths = tests
```
Comment (kjsanger) [inline tools/automation/gitlab_test/pytest.ini:2]: Nit: missing EOL

File: .gitlab_include/mirror.yml [hunk 2]
```diff
@@ -3,4 +3,4 @@ mirror:
     - schedules
 
   script:
-    - scripts/mirror_repo.sh -l $CI_REPOSITORY_URL -g $GITHUB_REPO -k $GITHUB_TOKEN -n $GITLAB_TOKEN_NAME -t $GITLAB_TOKEN -i $PROJECT_ID
\ No newline at end of file
+    - tools/automation/mirror_repo.sh -l $CI_REPOSITORY_URL -g $GITHUB_REPO -k $GITHUB_TOKEN -n $GITLAB_TOKEN_NAME -t $GITLAB_TOKEN -i $PROJECT_ID
```
Comment (kjsanger) [inline .gitlab_include/mirror.yml:6]: Nit: missing EOL

File: tools/automation/gitlab_test/tests/data/recipes/fail/1.0.0/run_test.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+if [[ -z $conda_test_variable ]]
+then
+  exit 1
+fi
```
Comment (kjsanger) [inline tools/automation/gitlab_test/tests/data/recipes/fail/1.0.0/run_test.sh:4]: Nit: missing EOL

File: tools/automation/gitlab_test/tests/data/recipes/success/1.0.0/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package:
+  name: success
+  version: 1.0.0
+
+build:
+  number: 0
+
+test:
+  commands:
+    - exit 0
```
Comment (kjsanger) [inline tools/automation/gitlab_test/tests/data/recipes/success/1.0.0/meta.yaml:10]: Nit: missing EOL

**Review by kjsanger — CHANGES_REQUESTED** [review]

**Comment by mksanger** [comment]
> What's the best way to run these locally? I've tried `cd`ing to `gitlab_test` and running `pytest`:

I'll add something to make this more clear, it needs to be run as a module, `python -m pytest`.

**Review by kjsanger — COMMENTED** [review]
I still get this `ModuleNotFoundError: No module named 'conda'` from the added dependency. You can add a `test-requirements.txt` at the top level (beside `requirements.txt`) to decalre that (and `pytest` too)

**Comment by kjsanger** [comment]
There's a few lines longer than PEP8 would like (PEP 8: E501 line too long (89 > 79 characters))

**Comment by kjsanger** [comment]
Fails with `No module named 'networkx'` so needs requirements fo that too.

**Comment by kjsanger** [comment]
Fails with `No module named 'networkx'` so needs requirements fo that too.

**Comment by kjsanger** [comment]
```
gitlab_test$ python -m pytest .
================================================================= test session starts =================================================================
platform linux -- Python 3.9.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/keith/dev/work/cloud/npg_conda/tools/automation/gitlab_test, configfile: pytest.ini
plugins: mock-3.6.1
collected 6 items / 1 error / 5 selected                                                                                                              

======================================================================= ERRORS ========================================================================
___________________________________________________________ ERROR collecting gitlab_test.py ___________________________________________________________
gitlab_test.py:34: in <module>
    args = parser.parse_args()
/home/keith/.local/miniconda/envs/npg_conda/lib/python3.9/argparse.py:1821: in parse_args
    self.error(msg % ' '.join(argv))
/home/keith/.local/miniconda/envs/npg_conda/lib/python3.9/argparse.py:2575: in error
    self.exit(2, _('%(prog)s: error: %(message)s\n') % args)
/home/keith/.local/miniconda/envs/npg_conda/lib/python3.9/argparse.py:2562: in exit
    _sys.exit(status)
E   SystemExit: 2
------------------------------------------------------------------- Captured stderr -------------------------------------------------------------------
usage: __main__.py [-h] [-b BUILD_DIR] [-c CHANGES] [-p PROD_CHANNEL]
                   [-d DEVEL_CHANNEL]
__main__.py: error: unrecognized arguments: .
================================================================== warnings summary ===================================================================
../../../../../../../.local/miniconda/envs/npg_conda/lib/python3.9/site-packages/conda/_vendor/auxlib/_vendor/five.py:49
  /home/keith/.local/miniconda/envs/npg_conda/lib/python3.9/site-packages/conda/_vendor/auxlib/_vendor/five.py:49: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    from imp import reload                  # noqa

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=============================================================== short test summary info ===============================================================
ERROR gitlab_test.py - SystemExit: 2
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================= 1 warning, 1 error in 0.70s =============================================================
```

**Comment by mksanger** [comment]
> `gitlab_test$ python -m pytest .`
I didn't mean to edit it... without the `.` should work, including that makes it try to run `gitlab_test.py`, which obviously fails due to not having any arguments.  Any ideas for a name for that script that doesn't include the word test?

**Pull request merged**